src/pal/src/exception/signal.cpp

```diff
@@ -283,6 +283,20 @@ static void sigsegv_handler(int code, siginfo_t *siginfo, void *context)
         record.ExceptionAddress = GetNativeContextPC(ucontext);
         record.NumberParameters = 2;
 
+        if (record.ExceptionCode == EXCEPTION_ACCESS_VIOLATION)
+        {
+            // Check if the failed access has hit a stack guard page. In such case, it
+            // was a stack probe that detected that there is not enough stack left.
+            void* stackLimit = CPalThread::GetStackLimit();
+            void* stackGuard = (void*)((size_t)stackLimit - getpagesize());
+            if ((siginfo->si_addr >= stackGuard) && (siginfo->si_addr < stackLimit))
+            {
+                // The exception happened in the page right below the stack limit,
+                // so it is a stack overflow
+                abort();
```

Comment:
You have just run `CPalThread::GetStackLimit()` here - so there is a bit of stack available. If you print it using a direct write to stderr (it should require very little stack - it should be just the system call), it should be about as reliable as the rest of this method.